examples/seismic/tutorials/09_viscoelastic.ipynb

```diff
@@ -0,0 +1,338 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Required imports:\n",
+    "import numpy as np\n",
+    "import sympy as sp\n",
+    "\n",
+    "from devito import *\n",
+    "from examples.seismic.source import RickerSource, TimeAxis\n",
+    "from examples.seismic import ModelViscoelastic, plot_image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Domain size:\n",
+    "extent = (200., 100., 100.)\n",
+    "h = 1.0\n",
+    "shape = (int(extent[0]/h+1), int(extent[1]/h+1), int(extent[2]/h+1))\n",
+    "\n",
+    "# Model physical parameters:\n",
+    "vp = np.zeros(shape)\n",
+    "qp = np.zeros(shape)\n",
+    "vs = np.zeros(shape)\n",
+    "qs = np.zeros(shape)\n",
+    "rho = np.zeros(shape)\n",
+    "\n",
+    "# Set up three horizontally separated layers:\n",
+    "for i in range(shape[0]):\n",
+    "    for j in range(shape[1]):\n",
+    "        for k in range(shape[2]):\n",
+    "            if k < int(0.5*extent[2]):\n",
+    "                vp[i, j, k] = 1.52\n",
+    "                qp[i, j, k] = 10000.\n",
+    "                vs[i, j, k] = 0.\n",
+    "                qs[i, j, k] = 0.\n",
+    "                rho[i, j, k] = 1.05\n",
+    "            elif int(0.5*extent[2]) <= k < int(0.5*extent[2]+4.*h):\n",
+    "                vp[i, j, k] = 1.6\n",
+    "                qp[i, j, k] = 40.\n",
+    "                vs[i, j, k] = 0.4\n",
+    "                qs[i, j, k] = 30.\n",
+    "                rho[i, j, k] = 1.3            \n",
+    "            else:\n",
+    "                vp[i, j, k] = 2.2\n",
+    "                qp[i, j, k] = 100.0\n",
+    "                vs[i, j, k] = 1.2\n",
+    "                qs[i, j, k] = 70.\n",
+    "                rho[i, j, k] = 2.0"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create model\n",
+    "origin = (0, 0, 0)\n",
+    "spacing = (extent[0]/(shape[0]-1),\n",
+    "           extent[1]/(shape[1]-1),\n",
+    "           extent[2]/(shape[2]-1))\n",
+    "so = 4 # FD space order\n",
+    "nbpml = 20 # Number of absorbing boundary layers cells\n",
+    "model = ModelViscoelastic(space_order=so, vp=vp, qp=qp, vs=vs, qs=qs,\n",
+    "                          rho=rho, origin=origin, shape=shape, spacing=spacing,\n",
+    "                          nbpml=nbpml)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Source freq. in MHz (note that the source is defined below):\n",
+    "f0 = 0.12\n",
+    "\n",
+    "# Thorbecke's parameter notation\n",
+    "cp2 = model.vp*model.vp\n",
+    "cs2 = model.vs*model.vs\n",
+    "ro = 1./model.rho\n",
+    "\n",
+    "mu = cs2*ro\n",
+    "l = (cp2*ro - 2*mu)\n",
```

Comment:
These are wrong, saw that doing vector branch, correct parameters are
```
ro = 1/rho
mu = cs2 * rho
l = (cp2*rho - 2*mu
```
it is the density for lambda/rho not its invers